Incorrect skip condition over `napari/_qt/_tests/test_app.py::test_windows_grouping_overwrite` and test fail when running on Windows

### 🐛 Bug Report

Working on PR #7269 noticed that there is a test which skip condition is always `False`:

https://github.com/napari/napari/blob/dfb4dd60a88046cbcd18acc6a38c901eeea2eed3/napari/_qt/_tests/test_app.py#L11-L12

However, changing the skip condition and then enabling the test to run on Windows (with something like `@pytest.mark.skipif(os.name != 'nt', reason='Windows specific')`), shows that the tests fails

### 💡 Steps to Reproduce

1. Run `napari/_qt/_tests/test_app.py::test_windows_grouping_overwrite` on Windows
2. See that the test is skipped
3. Change skip condition to use `nt` instead of `Windows` over the `os.name` validation
4. Run again the test
5. The test runs but it fails:

```python
>       assert get_app_id() == ''
E       AssertionError: assert 'custom_string' == ''
E
E         + custom_string

napari\_qt\_tests\test_app.py:46: AssertionError
```

### 💡 Expected Behavior

To be able to run `napari/_qt/_tests/test_app.py:test_windows_grouping_overwrite` on Windows successfully

### 🌎 Environment

napari: 0.5.4.dev19+gdfb4dd60
Platform: Windows-10-10.0.19045-SP0
Python: 3.10.14 | packaged by conda-forge | (main, Mar 20 2024, 12:40:08) [MSC v.1938 64 bit (AMD64)]
Qt: 5.15.2
PyQt5: 5.15.10
NumPy: 1.26.4
SciPy: 1.13.1
Dask: 2024.6.1
VisPy: 0.14.3
magicgui: 0.8.3
superqt: 0.6.7
in-n-out: 0.2.1
app-model: 0.2.9.dev1+ga47de13
npe2: 0.7.6

OpenGL:
  - GL version:  4.5.0 - Build 25.20.100.6444
  - MAX_TEXTURE_SIZE: 16384
  - GL_MAX_3D_TEXTURE_SIZE: 2048

Screens:
  - screen 1: resolution 1366x768, scale 1.0
  - screen 2: resolution 1920x1080, scale 1.0

Optional:
  - numba: 0.60.0
  - triangle not installed
  - napari-plugin-manager: 0.1.1

Settings path:
  - C:\Users\dalth\AppData\Local\napari\napari-dev_4454fdbf170edefe23827a4bf512389a6806af69\settings.yaml
Plugins:
  - napari: 0.5.4.dev19+gdfb4dd60 (81 contributions)
  - napari-console: 0.0.9 (0 contributions)
  - napari-ome-zarr: 0.6.1 (2 contributions)
  - napari-svg: 0.2.0 (2 contributions)

### 💡 Additional Context

_No response_

--- a/napari/_qt/_tests/test_app.py
+++ b/napari/_qt/_tests/test_app.py
@@ -8,7 +8,7 @@ from qtpy.QtWidgets import QAction, QShortcut
 from napari._qt.qt_event_loop import _ipython_has_eventloop, run, set_app_id
 
 
-@pytest.mark.skipif(os.name != 'Windows', reason='Windows specific')
+@pytest.mark.skipif(os.name != 'nt', reason='Windows specific')
 def test_windows_grouping_overwrite(qapp):
     import ctypes
 

--- a/napari/_qt/_tests/test_app.py
+++ b/napari/_qt/_tests/test_app.py
@@ -42,8 +42,10 @@ def test_windows_grouping_overwrite(qapp):
     assert get_app_id() == 'test_text'
     set_app_id('custom_string')
     assert get_app_id() == 'custom_string'
-    set_app_id('')
-    assert get_app_id() == ''
+    set_app_id('')  # app id can't be an empty string
+    assert get_app_id() == 'custom_string'
+    set_app_id(' ')
+    assert get_app_id() == ' '
 
 
 def test_run_outside_ipython(make_napari_viewer, qapp, monkeypatch):

